docs/src/design/many_differentials.md

```diff
@@ -1,113 +1,175 @@
+This is in-contrast to the Swift AD efforts, which has one differential type per primal type. (Swift uses the term associated tangent type, rather than differential type).
 
-This is in-contrast to the Swift AD efforts.
-Which has one differential type, per primal type.
-(Swift uses the term associated tangent type, rather than differential type).
+!!! terminology "differential and associated tangent type"
+    The use of “associated tangent type” in AD is not technically correct, as differentials naturally live in the [_cotangent_ plane](https://en.wikipedia.org/wiki/Cotangent_space) instead of the [tangent plane](https://en.wikipedia.org/wiki/Tangent_space). 
+    However it is often reasonable for AD to treat the cotangent plane and tangent plane as the same thing, and this was an intentional choice by the Swift team.
+    Here we will be a bit more tight with terminology and just stick to the ChainRules terminology and only say “differential type” instead of “tangent type”.
 
-One thing to understand about differentials is they have to form a [vector space](https://en.wikipedia.org/wiki/Vector_space) (or something very like them).
-They need to support addition to each other, they need a zero which doesn't change what it is added to, and they need to support scalar multiplication (this isn't really required, but its handy (e.g. for SGD).
-Beyond being a vector space, differentials need to be able to be added to a primal value to get back another primal value.
+One thing to understand about differentials is they have to form a [vector space](https://en.wikipedia.org/wiki/Vector_space)  (or something very like them).
+They need to support addition to each other, they need a zero which doesn't change what it is added to, and they need to support scalar multiplication (this isn't really required, but it is handy for things like gradient descent). 
+Beyond being a vector space, differentials need to be able to be added to a primal value to get back another primal value. 
 Or roughly equivalently a differential is a difference between two primal values.
 
-One thing to note in this example is that the *primal* does not have to be a vector space.
-We can however always transform it to be a vector space via choosing it an origin and expressing each primal value as a differential typed distance from that origin.
+One thing to note in this example is that the primal does not have to be a vector space.
+We can however always transform it to be a vector space via choosing an origin, and expressing each primal value as a differential typed distance from that origin.
+As an example, consider `DateTime`. A `DateTime` is not a vector space: there is no `zero(::DateTime)`, and `DateTime`s cannot be added to each other. The corresponding differential type is any subtype of `Period`, such as `Millisecond`, `Hour`, `Day` etc. 
+### Natural differential
```

Comment:
I'm curious about the motivation of natural differentials. Correct me if I'm wrong:
 ``X = V_1\times V_2 \times ... \times V_k``, where ``V_i`` are ``K_i``-vector spaces (``K_i`` are float types) are precisely  spaces ``X`` where natural differentials exist. In this case, we have the above torsor-like structure (except we have an additive action by this product of vector spaces!). The natural differential type is isomorphic to ``X``, only the identification between ``X`` and ``D(X)`` is ambiguous (changing to ``D``-notation).

Reply:
The motivation is solely around the fact that the type system exists and there are some types people want to use to represent differentials, that are not `Composite`.